ako-infra/avirest/handle_deconfig.go

```diff
@@ -0,0 +1,110 @@
+)
+
+func DeleteServiceEngines() error {
+	client := avicache.SharedAVIClients().AviClient[0]
+	segName := lib.GetClusterID()
+	cloudName := utils.CloudName
+
+	seListUri := "/api/serviceengine/?page_size=100&include_name&se_group_ref.name=" + segName + "&cloud_ref.name=" + cloudName
+	response := models.ServiceEngineAPIResponse{}
+	err := lib.AviGet(client, seListUri, &response)
+	if err != nil {
+		if aviError, ok := err.(session.AviError); ok && aviError.HttpStatusCode == 403 {
+			//SE in provider context no read access
+			utils.AviLog.Debugf("Switching to admin context from  %s", lib.GetTenant())
+			SetAdminTenant := session.SetTenant(lib.GetAdminTenant())
+			SetTenant := session.SetTenant(lib.GetTenant())
+			SetAdminTenant(client.AviSession)
+			defer SetTenant(client.AviSession)
+			if err := lib.AviGet(client, seListUri, &response); err != nil {
+				utils.AviLog.Warnf("Error during Get call for the SEs: %v", err.Error())
+				return err
+			}
+		} else {
+			utils.AviLog.Warnf("Error during Get call for the SEs: %v", err.Error())
+			return err
+		}
+	}
+
+	if len(response.Results) == 0 {
+		utils.AviLog.Infof("No service engines found for service engine group %s in cloud %s", segName, cloudName)
+		return nil
+	}
+
+	seUUIDs := []string{}
+	for _, se := range response.Results {
+		seUUIDs = append(seUUIDs, *se.UUID)
+	}
+
+	for _, seUUID := range seUUIDs {
+		deleteUri := "/api/serviceengine/" + seUUID
+		if err := lib.AviDelete(client, deleteUri); err != nil {
+			utils.AviLog.Errorf("Unable to delete service engine: %s, %s", seUUID, err.Error())
+			return err
+		}
+	}
+	return nil
+}
+
+func DeleteServiceEngineGroup() error {
+	client := avicache.SharedAVIClients().AviClient[0]
+	segName := lib.GetClusterID()
+	cloudName := utils.CloudName
+
+	segListUri := "/api/serviceenginegroup/?name=" + segName + "cloud_ref.name=" + cloudName
+	response := models.ServiceEngineGroupAPIResponse{}
+	err := lib.AviGet(client, segListUri, &response)
+	if err != nil {
+		if aviError, ok := err.(session.AviError); ok && aviError.HttpStatusCode == 403 {
+			//SE in provider context no read access
+			utils.AviLog.Debugf("Switching to admin context from  %s", lib.GetTenant())
+			SetAdminTenant := session.SetTenant(lib.GetAdminTenant())
+			SetTenant := session.SetTenant(lib.GetTenant())
+			SetAdminTenant(client.AviSession)
+			defer SetTenant(client.AviSession)
+			if err := lib.AviGet(client, segListUri, &response); err != nil {
+				utils.AviLog.Warnf("Error during Get call for the SE group :%v", err.Error())
+				return err
+			}
+		} else {
+			utils.AviLog.Warnf("Error during Get call for the SE group :%v", err.Error())
+			return err
+		}
+	}
+
+	if len(response.Results) == 0 {
+		utils.AviLog.Infof("No service engine groups found in cloud %s", cloudName)
+		return nil
+	}
+
+	segUuid := *response.Results[0].UUID
+	deleteSEGUri := "/api/serviceenginegroup/" + segUuid
```

Comment:
Instead of deleting the first SE from the list should we query by SEGroup Name and delete the specific SEGroup ?

Reply:
I am fetching the SEs using the se group name reference in the GET call above.

Reply:
missed it, thanks